lib/matplotlib/tests/test_artist.py

```diff
@@ -340,3 +340,38 @@ def func(artist):
     art.remove_callback(oid)
     art.pchanged()  # must not call the callback anymore
     assert func.counter == 2
+
+
+def test_set_signature():
+    """Test autogenerated ``set()`` for Artist subclasses."""
+    class MyArtist1(martist.Artist):
+        def set_myparam1(self, val):
+            pass
+
+    assert hasattr(MyArtist1.set, '_autogenerated_signature')
+    assert 'myparam1' in MyArtist1.set.__doc__
+
+    class MyArtist2(MyArtist1):
+        def set_myparam2(self, val):
+            pass
+
+    assert hasattr(MyArtist2.set, '_autogenerated_signature')
+    assert 'myparam1' in MyArtist2.set.__doc__
+    assert 'myparam2' in MyArtist2.set.__doc__
+
+
+def test_set_is_overwritten():
+    """set() defined in Artist subclasses should not be overwritten."""
+    class MyArtist3(martist.Artist):
+
+        def set(self, **kwargs):
+            """Not overwritten."""
+            pass
```

Comment:
(You can skip the `pass` (the docstring suffices as function body) and keep codecov *slightly* happier.)

Reply:
yup.